modules/lang-painless/src/doc/java/org/elasticsearch/painless/ContextApiSpecGenerator.java

```diff
@@ -70,11 +72,40 @@ private static Path resetRootDir() throws IOException {
     }
 
     @SuppressForbidden(reason = "resolve jdk src directory with environment")
-    private static Path getJdkSrc() {
+    private static JavaClassFilesystemResolver getJdkSrc() {
         String jdksrc = System.getProperty("jdksrc");
         if (jdksrc == null || "".equals(jdksrc)) {
             return null;
         }
-        return PathUtils.get(jdksrc);
+        return new JavaClassFilesystemResolver(PathUtils.get(jdksrc));
+    }
+
+    public static class JavaClassFilesystemResolver implements JavaClassResolver {
+        private final Path root;
+
+        public JavaClassFilesystemResolver(Path root) {
+            this.root = root;
+        }
+
+        @SuppressForbidden(reason = "resolve class file from java src directory with environment")
+        public InputStream openClassFile(String className) throws IOException {
+            // TODO(stu): handle primitives & not stdlib
```

Comment:
Don't forget about the todo here. Maybe move to an issue?

Reply:
Added https://github.com/elastic/elasticsearch/issues/68688